Jint/Engine.Modules.cs

```diff
@@ -28,6 +42,60 @@ internal JsModule LoadModule(string referencingModuleLocation, string specifier)
             return module;
         }
 
+        public JsModule DefineModule(string source, string specifier)
+        {
+            var module = new JavaScriptParser(source).ParseModule();
+
+            return DefineModule(module, specifier);
+        }
+
+        public JsModule DefineModule(Module source, string specifier)
+        {
+            var key = new ModuleCacheKey(string.Empty, specifier);
+
+            var module = new JsModule(this, _host.CreateRealm(), source, null, false);
+
+            _modules[key] = module;
+
+            return module;
+        }
+
+        public ObjectInstance ImportModule(string specifier)
+        {
+            var key = new ModuleCacheKey(string.Empty, specifier);
+
+            if (!_modules.TryGetValue(key, out var module))
+                throw new ArgumentOutOfRangeException(nameof(specifier), "No module was found for this specified");
```

Comment:
Typo on 'specified'. I assume wither 'specifier' or the actual name used.

Reply:
Fixed :)